flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobMetricsMessageParameters.java

```diff
@@ -22,19 +22,20 @@
 import java.util.Collections;
 
 /**
- * Message parameters which require a job path parameter.
+ * A job related REST handler always requires a {@link JobIDPathParameter}.
  */
-public class JobMessageParameters extends MessageParameters {
+public class JobMetricsMessageParameters extends MessageParameters {
```

Comment:
Why do you move this class? I think you should create a new sub class of `JobMessageParameters` which adds the `metricQueryParameter`.